back/engines/commercial/public_api/spec/acceptance/v2/comments_spec.rb

```diff
@@ -97,38 +97,8 @@
       end
     end
 
-    context "when filtering by 'created_at'" do
-      let(:created_at) { '2019-01-01,2019-01-10' }
-
-      let!(:returned_comments) do
-        comments.take(2).tap do |comments|
-          comments[0].update(created_at: '2019-01-01')
-          comments[1].update(created_at: '2019-01-10')
-        end
-      end
-
-      example_request 'List only the comments created between the specified dates' do
-        assert_status 200
-
-        comment_ids = json_response_body[:comments].pluck(:id)
-        expect(comment_ids).to match_array(returned_comments.pluck(:id))
-      end
-    end
-
-    context "when filtering by 'updated_at'" do
-      let(:updated_at) { '2021-01-01,2021-01-01' }
-
-      let!(:returned_comment) do
-        comments.first.tap { |comment| comment.update(updated_at: '2021-01-01') }
-      end
-
-      example_request 'List only the comments updated between the specified dates' do
-        assert_status 200
-
-        comment_ids = json_response_body[:comments].pluck(:id)
-        expect(comment_ids).to match_array([returned_comment.id])
-      end
-    end
+    include_examples 'filtering_by_date', :comment, :created_at
+    include_examples 'filtering_by_date', :comment, :updated_at
```

Comment:
Not the biggest fan of abstracting too many things in tests, especially not in acceptance. It works until it doesn't and we need to parameterize for some endpoint. Also risk of bugs in the abstractions instead of the actual implementation. Can def live with it.

Reply:
There is always a trade-off involved for this kind of thing. I think in this case it's worth it for two reasons: 
- It's really saving us a lot of lines of code and makes writing tests for new endpoints much faster.
- The filtering logic is abstracted in the `ApiController`. For me, that's a good sign that the test counterpart should be abstracted as well.